tool/src/cpp/config.rs

```diff
@@ -0,0 +1,44 @@
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct LibraryType {
+    pub name: String,
+    pub expr: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct LibraryConfig {
+    pub headers: Vec<String>,
+    pub span: LibraryType,
+    pub string_view: LibraryType,
+    pub unique_ptr: LibraryType,
+}
+
+impl LibraryConfig {
+    pub fn default() -> LibraryConfig {
+        LibraryConfig {
+            headers: vec![
+                "#include <stdint.h>".into(),
+                "#include <stddef.h>".into(),
+                "#include <stdbool.h>".into(),
+                "#include <algorithm>".into(),
+                "#include <memory>".into(),
+                "#include <optional>".into(),
```

Comment:
thought: a lot of these includes are needed by Diplomat itself regardless of how you customize it. Should we have the config only be for headers needed by the three customizations below?